flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/main/java/com/ververica/cdc/connectors/starrocks/sink/StarRocksMetadataApplier.java

```diff
@@ -119,7 +119,7 @@ private void applyAddColumn(AddColumnEvent addColumnEvent) {
                             .setColumnName(column.getName())
                             .setOrdinalPosition(-1)
                             .setColumnComment(column.getComment());
-            toStarRocksDataType(column, builder);
+            toStarRocksDataType(column, column.isPrimaryKey(), builder);
```

Comment:
not support to add primary key, so can just set `false` here

Reply:
In my understanding, it is not necessary to set all CHAR to VARCHAR if there is a primary key on the table. Only when CHAR is on the primary key, it needs to be converted to VARCHAR.
Actually, I'm not sure if I understand your question. If I answer incorrectly, could you please describe the problem again.

Reply:
I mean the added column can't be a primary key in StarRocks, so the flag `isPrimaryKey` should always be false here

Reply:
done